compio-driver/src/lib.rs

```diff
@@ -361,34 +354,15 @@ impl Entry {
     pub fn into_result(self) -> io::Result<usize> {
         self.result
     }
-}
 
-// The output entries need to be marked as `completed`. If an entry has been
-// marked as `cancelled`, it will be removed from the registry.
-struct OutEntries<'b, E> {
-    entries: &'b mut E,
-}
-
-impl<'b, E> OutEntries<'b, E> {
-    pub fn new(entries: &'b mut E) -> Self {
-        Self { entries }
-    }
-}
-
-impl<E: Extend<usize>> Extend<Entry> for OutEntries<'_, E> {
-    fn extend<T: IntoIterator<Item = Entry>>(&mut self, iter: T) {
-        self.entries.extend(iter.into_iter().filter_map(|e| {
-            let user_data = e.user_data();
-            let mut op = unsafe { Key::<()>::new_unchecked(user_data) };
-            op.set_flags(e.flags());
-            if op.set_result(e.into_result()) {
-                // SAFETY: completed and cancelled.
-                let _ = unsafe { op.into_box() };
-                None
-            } else {
-                Some(user_data)
-            }
-        }))
+    pub fn notify(self) {
```

Comment:
Either `Entry::new` should be unsafe or `Entry::notify` should be unsafe, or the creation of `Key` here will be unsound (you have to guarantee user data is a valid ptr to a valid `RawOp`). I prefer the latter.

Reply:
Also remember to add SAFETY justification whenever you write unsafe block. It reminds you to uphold the invariants.